mne/io/reference.py

```diff
@@ -298,6 +298,10 @@ def set_eeg_reference(inst, ref_channels=None, copy=True, verbose=None):
                 inst.info['custom_ref_applied'] = False
                 inst.add_proj(make_eeg_average_ref_proj(inst.info,
                               activate=False))
+                # Apply the reference if data has been preloaded
+                if hasattr(inst, '_data') or hasattr(inst, 'data'):
+                    inst.apply_proj()
```

Comment:
I'm afraid that would be a very confusing parameter to explain to users, as it would only do something in the case where you apply an average reference to data that has been preloaded. Pretty specific. I'll just print a message instead.